### ISSUE TITLE
Deprecate allYourBase and accumulateBases

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4946]
|[<img alt="lvh's avatar" src="https://avatars.githubusercontent.com/u/97816?s=50" width="50" height="50">](https://github.com/lvh)| @lvh reported|
|-|-|
|Trac ID|trac#4946|
|Type|enhancement|
|Created|2011-03-14 15:49:12Z|
|Branch|https://github.com/twisted/twisted/tree/deprecate-reflect-stuff-4946|

For more details see [#4928](https://github.com/twisted/twisted/issues/4928) -- we've removed the use cases for this except in some code which is going to be deprecated so we should probably deprecate this in favor of fancy inspect module functions.

Attachments:

* [4946.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/a02/a029f2021d08eb135d7ad51e9362d04b6212f1bf/5a74106a3b152517d054b94a9a12ff07ff80039e.patch) (2101 bytes) - added by aruse on 2011-03-14 18:41:21Z - Deprecation of allYourBase and accumulateBases

<details><summary>Searchable metadata</summary>

```
trac-id__4946 4946
type__enhancement enhancement
reporter__lvh lvh
priority__normal normal
milestone__ 
branch__branches_deprecate_reflect_stuff_4946 branches/deprecate-reflect-stuff-4946
branch_author__lvh lvh
status__closed closed
resolution__fixed fixed
component__core core
keywords__easy easy
time__1300117752000000 1300117752000000
changetime__1300136738000000 1300136738000000
version__None None
owner__lvh lvh

```
</details>

**Comment by twisted-trac** [comment]
|<img alt="aruse's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|aruse set owner to aruse<br>aruse set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="aruse's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|aruse removed owner<br>aruse set status to `new`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="lvh's avatar" src="https://avatars.githubusercontent.com/u/97816?s=50" width="50" height="50">](https://github.com/lvh)|@lvh set owner to @lvh<br>@lvh set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="lvh's avatar" src="https://avatars.githubusercontent.com/u/97816?s=50" width="50" height="50">](https://github.com/lvh)<a name="note_4"></a>|@lvh commented|
|-|-|

(In [31036]) Branching to 'deprecate-reflect-stuff-4946'

**Comment by twisted-trac** [comment]
|[<img alt="lvh's avatar" src="https://avatars.githubusercontent.com/u/97816?s=50" width="50" height="50">](https://github.com/lvh)<a name="note_5"></a>|@lvh commented|
|-|-|

Thanks for your patch! This is great, but blocking on [#4928](https://github.com/twisted/twisted/issues/4928) since until that gets merged Twisted itself still uses allYourBase so it fires its own DeprecationWarning. As soon as that's fixed we should merge branches/deprecate-reflect-stuff-4946 into trunk.

**Comment by twisted-trac** [comment]
|[<img alt="lvh's avatar" src="https://avatars.githubusercontent.com/u/97816?s=50" width="50" height="50">](https://github.com/lvh)<a name="note_6"></a>|@lvh set status to `closed`|
|-|-|

(In [31056]) allYourBase and accumulateBases are now deprecated in favor of inspect.getmro

Author: aruse
Reviewer: lvh
Fixes: [#4946](https://github.com/twisted/twisted/issues/4946)
